### PR TITLE
Added sdp examples 

### PR DESCRIPTION
**Pull request by Ayush-iitkgp**
From my experience, I found that posing a problem as Semi-Definite Programming is quite difficult for the newbies. Hence, I am trying to add more examples for the users to make themselves more comfortable with SDP. I have created a sub-directory inside examples folder especially dedicated to SDP examples so that users find it easy to read and understand. 

I will be adding more examples in the upcoming days as I identify which are the fields where SDP are being used.

### CODE REVIEW

**Comment by Ayush-iitkgp** [comment]
Ping @madeleineudell .

File: examples/sdp_examples/Converting a non-dcp compliant problem to Semi-Definite Programming Problem.ipynb [hunk 1]
```diff
@@ -0,0 +1,111 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#  Non-DCP Compliant Problem to Semi-Definite Programming Problem \n",
+    "\n",
```
Comment (madeleineudell) [inline examples/sdp_examples/Converting a non-dcp compliant problem to Semi-Definite Programming Problem.ipynb]: I'd suggest adding a bit of intro here: what's the idea of what you're doing below? For what kinds of problems should a newbie expect it to work?

File: examples/sdp_examples/Converting a non-dcp compliant problem to Semi-Definite Programming Problem.ipynb [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+    "Consider the problem\n",
+    "\n",
+    "minimize $\\frac{(c^T x)^2}{d^Tx}$\n",
+    "\n",
+    "subject to Ax +b$\\geq$ 0\n",
+    "\n",
+    "Also $d^Tx$>0 whenever Ax+b$\\geq$ 0\n",
```
Comment (madeleineudell) [inline examples/sdp_examples/Converting a non-dcp compliant problem to Semi-Definite Programming Problem.ipynb]: Make it clear that this is an *assumption*.

File: examples/sdp_examples/Converting a non-dcp compliant problem to Semi-Definite Programming Problem.ipynb [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+    "\n",
+    "**It is very easy to obseve that the objective function is neither concave or convex or affine and hence it is not dcp compliant problem.**\n",
```
Comment (madeleineudell) [inline examples/sdp_examples/Converting a non-dcp compliant problem to Semi-Definite Programming Problem.ipynb]: typo: observe

File: examples/sdp_examples/Converting a non-dcp compliant problem to Semi-Definite Programming Problem.ipynb [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+    "\n",
+    "** exp = square(c'*x)/d'*x **    \n",
```
Comment (madeleineudell) [inline examples/sdp_examples/Converting a non-dcp compliant problem to Semi-Definite Programming Problem.ipynb]: The formatting of this whole section is weird. Did you run the code in the .ipynb or copy-paste it from a terminal? The former method is preferred.

File: examples/sdp_examples/Converting a non-dcp compliant problem to Semi-Definite Programming Problem.ipynb [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+    "\n",
+    "## Equivalent SDP\n",
+    "\n",
+    "We introduce a new variable $t \\in \\mathbf{R}$\n",
+    "\n",
+    "Now define $D$ as follows:\n",
+    "\n",
+    "$D=\\left[\\begin{array}{cc}t&c^Tx\\\\c^Tx&d^Tx\\end{array}\\right]$\n",
+    "\n",
+    "Using the theory of Schur Complements we can show that $D \\succeq 0$ (i.e is D is positive semi-definite matrix)\n",
```
Comment (madeleineudell) [inline examples/sdp_examples/Converting a non-dcp compliant problem to Semi-Definite Programming Problem.ipynb]: ...only for some t's! That's the most important part of the argument...! Spend a few more sentences explaining this.

Comment (Ayush-iitkgp) [inline examples/sdp_examples/Converting a non-dcp compliant problem to Semi-Definite Programming Problem.ipynb]: Yes. I would incorporate your suggestions .

File: examples/sdp_examples/Converting a non-dcp compliant problem to Semi-Definite Programming Problem.ipynb [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#  Non-DCP Compliant Problem to Semi-Definite Programming Problem \n",
+    "\n",
+    "Consider the problem\n",
+    "\n",
+    "minimize $\\frac{(c^T x)^2}{d^Tx}$\n",
```
Comment (madeleineudell) [inline examples/sdp_examples/Converting a non-dcp compliant problem to Semi-Definite Programming Problem.ipynb]: I'd suggest writing optimization problems with equation arrays in latex:

\begin{array}{ll}
\mbox{minimize} & \frac{(c^T x)^2}{d^Tx} \\
\mbox{subject to} & \ldots
\end{array}

**Comment by madeleineudell** [comment]
@Ayush-iitkgp , have you had a chance to clean up this example so we can merge it?

**Comment by Ayush-iitkgp** [comment]
@madeleineudell, my bad, I seem to have forgotten the source from where I got this example.
I have made the commits as per your suggestions. See if we can merge this into the master.

**Comment by odow** [comment]
Closing as stale. This hasn't been touched for 5 years.

**Pull request closed without merging**